lib/auth/auth_with_roles.go

```diff
@@ -343,30 +343,53 @@ func (a *ServerWithRoles) Export(ctx context.Context, req *collectortracepb.Expo
 		}
 	}
 
-	forwardedFor := sb.String()
+	// the forwarded attribute to add
+	value := &otlpcommonv1.KeyValue{
+		Key: forwardedTag,
+		Value: &otlpcommonv1.AnyValue{
+			Value: &otlpcommonv1.AnyValue_StringValue{
+				StringValue: sb.String(),
+			},
+		},
+	}
+
+	// returns true if the attributes don't already contain
+	// the forwardedTag
```

Comment:
This makes it so that a malicious user can just submit a fake `forwardedTag` and we'll honor it. We should instead go through `Attributes`, replace any with the `forwardedTag`, and if there's none, add a new one.